warehouse/query-core/src/main/java/datawave/query/jexl/nodes/QueryPropertyMarker.java

```diff
@@ -49,7 +49,7 @@ public static String label() {
      *            the {@link QueryPropertyMarker} type
      * @return a new marker instance
      */
-    public static <MARKER extends QueryPropertyMarker> MARKER create(JexlNode source, Function<JexlNode,MARKER> constructor) {
+    public static <MARKER extends QueryPropertyMarker> JexlNode create(JexlNode source, Function<JexlNode,MARKER> constructor) {
```

Comment:
I would like this to return a MARKER rather than a JexlNode. In this method we're able to be specific about the type of node we're returning (since we're not potentially returning the original source node), so why not be? Alternatively, if we want to verify the source node isn't already marked for every marker type we currently have, then perhaps this method should be deleted in favor of `create(source, class)`.

Reply:
I'll revert this change along with any supporting changes necessary. While I would like to check the source node for every create, that may be overkill in which case having an alternate create method that is faster would be nice.